adam/perception/developmental_primitive_perception.py

```diff
@@ -0,0 +1,219 @@
+from abc import ABC
+
+from attr import attrib, attrs
+from attr.validators import in_, instance_of
+from immutablecollections import ImmutableSet, immutableset
+from immutablecollections.converter_utils import _to_immutableset
+from vistautils.range import Range
+
+from adam.ontology import Ontology, OntologyNode
+from adam.ontology.phase1_ontology import RECOGNIZED_PARTICULAR
+from adam.perception import PerceptualRepresentationFrame
+
+
+@attrs(slots=True, frozen=True, repr=False)
+class DevelopmentalPrimitivePerceptionFrame(PerceptualRepresentationFrame):
+    r"""
+    A static snapshot of a `Situation` based on developmentally-motivated perceptual primitives.
+
+    This represents a situation as
+
+    - a set of `ObjectPerception`\ s, with one corresponding to each
+      object in the scene (e.g. a ball, Mom, Dad, etc.)
+    - a set of `ObjectPerception`\ s which associate a `ObjectPerception` with perceived properties
+      of various sort (e.g. color, sentience, etc.)
+    - a set of `RelationPerception`\ s which describe the learner's perception of how two
+      `ObjectPerception`\ s are related.
+
+    This is the default perceptual representation for at least the first phase of the ADAM project.
+    """
+    perceived_objects: ImmutableSet["ObjectPerception"] = attrib(
+        converter=_to_immutableset, default=immutableset()
+    )
+    r"""
+    a set of `ObjectPerception`\ s, with one corresponding to each
+      object in the scene (e.g. a ball, Mom, Dad, etc.)
+    """
+    property_assertions: ImmutableSet["PropertyPerception"] = attrib(
+        converter=_to_immutableset, default=immutableset()
+    )
+    r"""
+    a set of `ObjectPerception`\ s which associate a `ObjectPerception` with perceived properties
+      of various sort (e.g. color, sentience, etc.)
```

Comment:
Should the documentation here be "a set of 'PropertyPerception'" instead?

Reply:
Good catch, fixed in c345a2d